### ISSUE TITLE
Linter failing to read angular.json after v16 migration

### ISSUE THREAD

**Issue by Blunderchips** [angular-eslint/angular-eslint#1366]
Reopening [this issue](https://github.com/angular/angular/issues/50202) here. I see the dependency is from https://github.com/angular-eslint.

**Description and reproduction of the issue**

As of Angular v16, the linter is failing to run. Not failing to pass but entirely failing to run. From the error message, it looks to me, it is looking for the `angular.json` in the wrong directory. The linter does run and pass on v15. Interestingly, `ng test` is still working as expected.

When running `ng lint`: (**file not found is to be expected as that is not where the `angular.json` is**)

![image](https://user-images.githubusercontent.com/7081037/236771621-39ed14f6-c24a-4343-b1e8-6843aa11f3b3.png)

The contents of the error file are:

![image](https://user-images.githubusercontent.com/7081037/236772061-6f62e808-41ca-4fbb-9754-0abe57dc996a.png)

**Versions**

```true
Angular CLI: 16.0.0                                           
Node: 18.15.0                                                 
Package Manager: npm 9.5.0                                    
OS: win32 x64                                                 
                                                              
Angular: 16.0.0                                               
... animations, cdk, cli, common, compiler, compiler-cli, core
... forms, localize, material, platform-browser               
... platform-browser-dynamic, router, service-worker          
                                                              
Package                         Version                       
---------------------------------------------------------     
@angular-devkit/architect       0.1600.0
@angular-devkit/build-angular   16.0.0
@angular-devkit/core            16.0.0
@angular-devkit/schematics      16.0.0
@angular/fire                   7.5.0
@angular/flex-layout            14.0.0-beta.41
@schematics/angular             16.0.0
ng-packagr                      16.0.0
rxjs                            7.8.1
typescript                      5.0.4
```

The project is a yarn berry mono-repo, it looks like the CLI is only looking in the root folder. Failing on both my local machine (windows 10) and out CI/CD pipeline (node 16 alpine).

**Comment by JamesHenry** [comment]
Hi @Blunderchips sorry that you are encountering an issue here.

Please could you add a log in your node_modules for me:

- Open the file: `node_modules/@angular-eslint/builder/dist/lint.impl.js`
- Add a the following line anywhere after the line `const systemRoot = context.root;`

```
console.log({ workspaceRoot: devkit_1.workspaceRoot, systemRoot });
```

And run `ng lint` again, and let me know what the values are please

**Comment by Blunderchips** [comment]
Hello @JamesHenry, thanks for getting back to me. I added the console here (screenshot below) but I am afraid that nothing else is being logged out. Seems to be failing before it hits that line.

![image](https://github.com/angular-eslint/angular-eslint/assets/7081037/785dcf48-4e75-4d08-8759-857a90063497)

**Comment by JamesHenry** [comment]
@Blunderchips Ok thanks.

Are you using PnP with yarn berry, or the node_modules linker?

**Comment by Blunderchips** [comment]
> @Blunderchips Ok thanks.
> 
> Are you using PnP with yarn berry, or the node_modules linker?

node_modules, not sure if this helps:

![image](https://github.com/angular-eslint/angular-eslint/assets/7081037/da6f126b-d236-4a79-b549-f3e273bf6426)

**Comment by JamesHenry** [comment]
@Blunderchips I cannot reproduce with just the yarn version and node-modules, so maybe the plugins you are using are to blame.

Please kindly provide the raw contents of the file for me to apply, your image is missing part of the URL on line 7

**Comment by JamesHenry** [comment]
Ok nvm, I think I found it by googling for the plugin name. I cannot reproduce the issue with this setup. The only obvious difference based on the information available at this point is I am on macOS.

I have pushed the repo in which I fail to reproduce, please kindly clone it and see if you have the issue and if need not please figure out what is different about it vs your original repo:

https://github.com/JamesHenry/angulareslintyarntest

**Comment by Blunderchips** [comment]
No, the issue was not present in the repo you created. I have put through a pull request that replicates it. I think the issue stems from the `angular.json` not being in the root directory.

https://github.com/JamesHenry/angulareslintyarntest/pull/2

**Comment by Blunderchips** [comment]
Hi @JamesHenry, do you need anything else from my response?

**Comment by runyasak** [comment]
Hi, I'm encountering this issue too.

Looking for the solution. 😁

**Comment by Blunderchips** [comment]
Same issue seems to be happening here https://github.com/nrwl/nx/issues/16998

@JamesHenry any indication of this is nx side or with angular-eslint?

**Comment by z3ttee** [comment]
I'm having the same issue. But for me it occurs in a monorepo using turborepo. A temporary fix is to downgrade angular-eslint to v15, but this is not ideal of course

**Comment by JamesHenry** [comment]
Am I right in saying that you are all experiencing when:

- using npm/yarn/pnpm workspaces
- AND the angular.json is located in a nested location, ie it is one of the workspaces within the repo, not at the root of the repo

**Comment by tpischke** [comment]
I can confirm the problem with yarn workspaces and angular 16

**Comment by Blunderchips** [comment]
> I can confirm the problem with yarn workspaces and angular 16
> 
> Workaround is to downgrade angular-eslint/builder to v15

Afraid that is not working for me (also using yarn workspaces). Going to wait until this is resolved before upgrading to Angular v16.

**Comment by JamesHenry** [comment]
Thanks all, this is now resolved in `16.0.2`

**Comment by Blunderchips** [comment]
Can, happily, confirm that all is working on my end. Thanks @JamesHenry!
